crates/codegen/parser/runtime/src/cst.rs

```diff
@@ -46,63 +46,103 @@ impl Node {
         }
     }
 
-    pub fn create_cursor(&self, text_offset: TextIndex) -> Cursor {
+    /// Creates a [`Cursor`] that starts at the current node as the root and a given initial `text_offset`.
+    pub fn cursor_with_offset(&self, text_offset: TextIndex) -> Cursor {
         Cursor::new(self.clone(), text_offset)
     }
 
+    /// Reconstructs the original source code from the parse tree.
```

Comment:
thoughts about using `.unparse()` instead of `.reconstruct()`?

https://en.wikipedia.org/wiki/Unparser

Reply:
Done in e6421cc6cf58aa52beb2abc8f92bc424af021ae4

Reply:
I made a comment about this elsewhere in this PR.